File: scripts/scip/fixtures/typescript/handler.ts
import { TaskService } from "./service";
import { Priority } from "./model";

interface CreateResult {
  status: string;
  task: string;
}

interface ListResult {
  tasks: string;
  pending: number;
}

export function handleCreate(title: string, priorityStr: string): CreateResult {
  const svc = new TaskService();
  const priority = priorityStr as Priority;
  const task = svc.addTask(title, priority);
  return { status: "created", task: `${task.title} (${task.priority})` };
}

export function handleList(): ListResult {
  const svc = new TaskService();
  return { tasks: svc.summary(), pending: svc.getPending().length };
}

File: scripts/scip/fixtures/python/handler.py
"""Request handler using TaskService."""

from service import TaskService
from model import Priority


def handle_create(title: str, priority_str: str) -> dict:
    svc = TaskService()
    priority = Priority(priority_str)
    task = svc.add_task(title, priority)
    return {"status": "created", "task": task.display()}


def handle_list() -> dict:
    svc = TaskService()
    return {"tasks": svc.summary(), "pending": len(svc.get_pending())}


if __name__ == "__main__":
    result = handle_create("Write tests", "high")
    print(result)
